### PR TITLE
[Presto-on-Spark Native] Propagate errors from Velox

### PR DESCRIPTION
**Pull request by mbasmanova**
Check TaskInfo for failures and propagate these.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-spark-base/src/main/java/com/facebook/presto/spark/execution/operator/NativeExecutionOperator.java [hunk 1]
```diff
@@ -160,37 +165,51 @@ public Page getOutput()
             checkState(process != null, "process is null");
             createTask();
             checkState(task != null, "task is null");
-            taskStatusFuture = task.start();
+            TaskInfo taskInfo = task.start();
```
Comment (shrinidhijoshi) [inline presto-spark-base/src/main/java/com/facebook/presto/spark/execution/operator/NativeExecutionOperator.java:168]: Should we ideally move this to `noMoreSplits`, or someplace we know is going to called once, and after all splits are added to operator

File: presto-spark-base/src/main/java/com/facebook/presto/spark/execution/operator/NativeExecutionOperator.java [hunk 2]
```diff
@@ -160,37 +165,51 @@ public Page getOutput()
...
+        if (taskStatus.getState() != TaskState.FINISHED) {
+            RuntimeException failure = taskStatus.getFailures().stream()
+                    .findFirst()
+                    .map(ExecutionFailureInfo::toException)
+                    .orElse(new PrestoException(GENERIC_INTERNAL_ERROR, "Native task failed for an unknown reason"));
+            throw failure;
+        }
+
+        info.set(new NativeExecutionInfo(ImmutableList.of(taskInfo.getStats())));
+        finished = true;
```
Comment (shrinidhijoshi) [inline presto-spark-base/src/main/java/com/facebook/presto/spark/execution/operator/NativeExecutionOperator.java:209]: Shouldn't we be setting `finished=true` even in the `!FINISHED` case above ?

File: presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionTask.java [hunk 3]
```diff
@@ -162,20 +162,30 @@ public void stop()
...
+                return response.getValue();
+            }
+            else {
+                throw new IllegalStateException("Create-or-update task request didn't return a result");
+            }
+        }
+        catch (InterruptedException e) {
+            throw new RuntimeException(e);
+        }
+        catch (ExecutionException e) {
```
Comment (shrinidhijoshi) [inline presto-spark-base/src/main/java/com/facebook/presto/spark/execution/NativeExecutionTask.java:186]: Nit: Can collapse into single catch block

File: presto-spark-base/src/main/java/com/facebook/presto/spark/execution/operator/NativeExecutionOperator.java (hunk 2 above)
Comment (mbasmanova) [reply presto-spark-base/src/main/java/com/facebook/presto/spark/execution/operator/NativeExecutionOperator.java:209]: I believe 'finished' flag is only for happy path.

File: presto-spark-base/src/main/java/com/facebook/presto/spark/execution/operator/NativeExecutionOperator.java (hunk 1 above)
Comment (mbasmanova) [reply presto-spark-base/src/main/java/com/facebook/presto/spark/execution/operator/NativeExecutionOperator.java:168]: We can do that, but probably in a separate PR.

Comment (shrinidhijoshi) [reply presto-spark-base/src/main/java/com/facebook/presto/spark/execution/operator/NativeExecutionOperator.java:168]: Actually, It should be feasible to move this whole block in `noMoreSplits()`  
```Java
if (process == null) {
    createProcess();
    checkState(process != null, "process is null");
    createTask();
    checkState(task != null, "task is null");
    TaskInfo taskInfo = task.start();
    if (processTaskInfo(taskInfo)) {
       return null;
    }
}
```

**Comment by shrinidhijoshi** [comment]
@mbasmanova  Thanks for fixing this one. LGTM

**Review by gggrace14 — APPROVED** [review]

**Pull request merged**